hail/src/main/scala/is/hail/expr/ir/PruneDeadFields.scala

```diff
@@ -321,6 +321,9 @@ object PruneDeadFields {
     BindingEnv(e, Some(e), Some(e))
   }
 
+  def uses(name: String, env: Env[BoxedArrayBuilder[Type]]): Array[Type] =
+    env.lookupOption(name).map(_.result()).getOrElse(Array.empty)
+
```

Comment:
This pattern was used a lot in this file. I took the oportunity to define this function and substitute it in. It's made the code cleaner at the expense of the review. Plz forgive.